cmd/tailscaled/tailscaled.go

```diff
@@ -96,6 +97,53 @@ var subCommands = map[string]*func([]string) error{
 	"debug":                   &debugModeFunc,
 }
 
+// Tailscale 1.14 and before stored state in
+// C:\WINDOWS\system32\config\systemprofile\AppData\Local
+// but it is frequently cleared for almost any reason: Windows
+// updates, System Restore, even various System Cleaner utilities.
+//
+// We carefully move any state in AppData\Local to ProgramData
+// at startup. If moving the state fails, we continue to use
+// AppData\Local so at least there will be connectivity.
+func tryWindowsAppDataMigration() {
+	if runtime.GOOS != "windows" {
+		return
+	}
+
+	oldDir := filepath.Join(os.Getenv("LocalAppData"), "Tailscale")
+	_, err := os.Stat(oldDir)
+	if os.IsNotExist(err) {
+		// Common case.
+		return
+	}
+	if err != nil {
+		log.Printf("tryWindowsAppDataMigration failed; cannot stat AppDataLocal: %v", err)
```

Comment:
"tryWindowsAppDataMigration failed: %v", err

... is probably sufficient.

Reply:
Done.